### PR TITLE
Feature/context logging

### PR DESCRIPTION
**Pull request by Fraise**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Various IDEs configs
+.idea/
+.vscode/
+
```
Comment (henyxia) [inline .gitignore:4]: IDE files should not be ignored in the project configuration but in the user global ignore file

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+#Executable
+open-bastion
+
+# Test binary, built with `go test -c`
+*.test
```
Comment (henyxia) [inline .gitignore:9]: Newline is missing

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+all: test build
+
+build:
+	go build $$(pwd)/cmd/open-bastion
```
Comment (henyxia) [inline Makefile:4]: Relative path is supported for the `go build` command. No need to invoke `pwd`

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+test:
+	go test ./...
```
Comment (henyxia) [inline Makefile:7]: Newline missing

File: internal/config/config.go [hunk 5]
```diff
@@ -3,11 +3,19 @@ package config
 import (
 	"encoding/json"
 	"errors"
+	logger "github.com/open-bastion/open-bastion/internal/logger"
```
Comment (henyxia) [inline internal/config/config.go:6]: renaming is useless here

File: internal/egress/egress.go [hunk 6]
```diff
@@ -1,65 +1,133 @@
...
+	logger.UpdateClientLogCtx(ctx, client)
+
+	if err != nil {
+		_, _ = client.SshCommChan.Write([]byte("error accessing credentials"))
+
+		logger.ErrorWithCtxWithErr(ctx, err, "authenticated user could not access his egress private key")
+	}
+
+	//TODO add proper cancel func to clean up and display an error
+	ctx, cancel := context.WithTimeout(ctx, time.Millisecond*1000)
```
Comment (henyxia) [inline internal/egress/egress.go:34]: 1 second is way too low, we might want to increase it and even make this a parameter

File: internal/obclient/client.go [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+type BackendConn struct {
+	Command string
+	User    string
+	Host    string
+	Port    int
+}
+
+//HandshakeSSH takes an initialized SSH configuration and tries to establish a SSH connection for the client.
+func (client *Client) HandshakeSSH(sshConfig *ssh.ServerConfig) error {
+	// func handshakeSSH(c *net.Conn, sshConfig *ssh.ServerConfig) (*ssh.ServerConn, <-chan ssh.NewChannel, error) {
```
Comment (henyxia) [inline internal/obclient/client.go:55]: forgotten dead code?

File: internal/egress/egress.go (hunk 6 above)
Comment (Fraise) [reply internal/egress/egress.go:34]: Will be done in a future PR!

**Pull request merged**